### PR TITLE
Fixing autoupdates for issues #5 #93

### PR DESCRIPTION
**Pull request by pathurs**

### CODE REVIEW

File: svo (install the zip, not me).xml [hunk 1]
```diff
@@ -54937,7 +54937,7 @@ end
 -- downloads the system &amp; updates the system version saved
 function svo.downloadnewsystem(newversion)
   svo.downloadedsystem = downloadfolder..string.format(&quot;%s svo.zip&quot;, svo.me.name)
-  local downloadurl = [[http://vadisystems.com/svo-r/r/]]..svo.me.name..[[%20svo.zip]]
+  local downloadurl = [[http://svof.pathurs.com/]]..svo.me.class..[[%20svo.zip]]
```
Comment (keneanung) [inline svo (install the zip, not me).xml]: The file name will be `<Class>.Svof.v<version>.zip` So this won't download the correct file.

**Comment by keneanung** [comment]
It also tries to access svo.me.name before login, which is not set yet due to the fact that svof is not bound to a character. This leads to an error. Instead svo.me.class should be used as the zip savepath.

File: svo (install the zip, not me).xml (hunk 1 above)
Comment (pathurs) [inline svo (install the zip, not me).xml]: I will try to find all references now and change to for example: jester.svof.v8.zip

File: svo (install the zip, not me).xml [hunk 2]
```diff
@@ -55040,28 +55040,28 @@ function svo_doupdate_click()
...
-  local name = svo.me.name
+  local class = svo.me.class
 
   if not (uninstallPackage and installPackage) then
     svo.updatelabel:echo([[&lt;p align=&quot;center&quot; style=&quot;font-size:10pt; color:white&quot;&gt;Can't update your Svo. You didn't update to Mudlet 2.1, and your current Mudlet lacks the features necessary. Go update first please.&lt;p&gt;]])
     return
   end
 
-  if not io.exists(getMudletHomeDir()..&quot;/svo/downloads/&quot;..string.format(&quot;%s svo.zip&quot;, svo.me.name)) then
+  if not io.exists(getMudletHomeDir()..&quot;/svo/downloads/&quot;..string.format(&quot;%s.Svof.current.zip&quot;, svo.me.class)) then
```
Comment (keneanung) [inline svo (install the zip, not me).xml]: `svo.me.class` got localized just a few lines further up. Maybe use it as well?

File: svo (install the zip, not me).xml [hunk 3]
```diff
@@ -55040,28 +55040,28 @@ function svo_doupdate_click()
...
+    uninstallPackage(class..&quot; svo&quot;)
 
     svo_updating_system = &quot;installing&quot;
     svo.updatelabel:echo([[&lt;p align=&quot;center&quot; style=&quot;font-size:10pt; color:white&quot;&gt;Uninstalled the old Svo, installing the new one now...&lt;p&gt;]])
-    installPackage(getMudletHomeDir()..&quot;/svo/downloads/&quot;..string.format(&quot;%s svo.zip&quot;, svo.me.name))
+    installPackage(getMudletHomeDir()..&quot;/svo/downloads/&quot;..string.format(&quot;%s.Svof.current.zip&quot;, svo.me.class))
 
     svo_updating_system = nil
-    svo.echof(&quot;Svo updated! Restart Mudlet, and you'll be all done. You can find the changelog on svo.vadisystems.com.&quot;)
+    svo.echof(&quot;Svo updated! Restart Mudlet, and you'll be all done. You can find the changelog on svof.pathurs.com.&quot;)
```
Comment (keneanung) [inline svo (install the zip, not me).xml]: http://svof.pathurs.com redirects to https://svof.github.io/svof which holds no changelog.

File: svo (install the zip, not me).xml [hunk 4]
```diff
@@ -55040,28 +55040,28 @@ function svo_doupdate_click()
...
 
   function svo_update_system()
     svo_updating_system = &quot;uninstalling&quot;
-    uninstallPackage(name..&quot; svo&quot;)
+    uninstallPackage(class..&quot; svo&quot;)
 
     svo_updating_system = &quot;installing&quot;
     svo.updatelabel:echo([[&lt;p align=&quot;center&quot; style=&quot;font-size:10pt; color:white&quot;&gt;Uninstalled the old Svo, installing the new one now...&lt;p&gt;]])
-    installPackage(getMudletHomeDir()..&quot;/svo/downloads/&quot;..string.format(&quot;%s svo.zip&quot;, svo.me.name))
+    installPackage(getMudletHomeDir()..&quot;/svo/downloads/&quot;..string.format(&quot;%s.Svof.current.zip&quot;, svo.me.class))
```
Comment (keneanung) [inline svo (install the zip, not me).xml]: Same deal with the localized class

File: svo (install the zip, not me).xml [hunk 5]
```diff
@@ -55090,9 +55090,9 @@ function svo_doupdate_click()
     if not svo_updating_system then return end
 
     if svo_updating_system == &quot;uninstalling&quot; then
-      svo.updatelabel:echo([[&lt;p align=&quot;center&quot; style=&quot;font-size:8pt; color:white&quot;&gt;Hm, seems the update got stuck on the uninstallPackage() function. Try uninstalling Svo from the Package Manager and then installing it from the zip (found in ]].. getMudletHomeDir()..&quot;/svo/downloads/&quot;..string.format(&quot;%s svo.zip&quot;, svo.me.name) ..[[) to upgrade now.&lt;p&gt;]])
+      svo.updatelabel:echo([[&lt;p align=&quot;center&quot; style=&quot;font-size:8pt; color:white&quot;&gt;Hm, seems the update got stuck on the uninstallPackage() function. Try uninstalling Svo from the Package Manager and then installing it from the zip (found in ]].. getMudletHomeDir()..&quot;/svo/downloads/&quot;..string.format(&quot;%s.Svof.current.zip&quot;, svo.me.class) ..[[) to upgrade now.&lt;p&gt;]])
```
Comment (keneanung) [inline svo (install the zip, not me).xml]: Same deal with the localized class

File: svo (install the zip, not me).xml [hunk 6]
```diff
@@ -55090,9 +55090,9 @@ function svo_doupdate_click()
...
     elseif svo_updating_system == &quot;installing&quot; then
-      svo.updatelabel:echo([[&lt;p align=&quot;center&quot; style=&quot;font-size:8pt; color:white&quot;&gt;Hm, seems the update got stuck on the installPackage() function. Old Svo is uninstalled, but it the new one didn't get installed - install your Svo zip (found in ]].. getMudletHomeDir()..&quot;/svo/downloads/&quot;..string.format(&quot;%s svo.zip&quot;, svo.me.name) ..[[ via the Package Manager.&lt;p&gt;]])
+      svo.updatelabel:echo([[&lt;p align=&quot;center&quot; style=&quot;font-size:8pt; color:white&quot;&gt;Hm, seems the update got stuck on the installPackage() function. Old Svo is uninstalled, but it the new one didn't get installed - install your Svo zip (found in ]].. getMudletHomeDir()..&quot;/svo/downloads/&quot;..string.format(&quot;%s.Svof.current.zip&quot;, svo.me.class) ..[[ via the Package Manager.&lt;p&gt;]])
```
Comment (keneanung) [inline svo (install the zip, not me).xml]: Same deal with the localized class

**Comment by keneanung** [comment]
@vadi2 I'm not sure the github page helps the average user. Especially since you need to scroll a lot until you find the tiny link to the docs.

**Comment by vadi2** [comment]
I'd argue the average person has more need to download the system than to look at the help manual which they never do, but I'm not too fussed on where you take this.

**Comment by keneanung** [comment]
@pathurs where do those two urls redirect to now?

**Comment by pathurs** [comment]
Wherever you want, I set homepage to the homepage you said on skype, and the changelog to the releases

**Comment by keneanung** [comment]
Could you explictely list them so @vadi2 can give us the go?

The only issue I have with it, is that devs get nagged about updates, but that can be fixed later.

**Comment by pathurs** [comment]
Homepage => svof.pathurs.com/homepage.php => https://github.com/svof/svof/#svof
Changelog => svof.pathurs.com/changelog.php => https://github.com/svof/svof/releases

**Comment by vadi2** [comment]
+1

**Comment by keneanung** [comment]
Thanks!

**Pull request merged**